androidApp/src/main/java/com/projectnewm/di/UseCaseModule.kt

```diff
@@ -0,0 +1,31 @@
+package com.projectnewm.di
+
+import com.projectnewm.example.ExampleRepository
+import com.projectnewm.example.ExampleService
+import com.projectnewm.example.ExampleUseCase
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object UseCaseModule {
+
+    @Singleton
+    @Provides
+    fun provideExampleUseCase(
+        exampleRepository: ExampleRepository,
+    ): ExampleUseCase {
+        return ExampleUseCase(exampleRepository)
+    }
+
+    @Singleton
+    @Provides
+    fun provideExampleRepository(
```

Comment:
Repository should probably be provided by a separate module but I'd like to refactor this anyway once we move everything into feature modules